prawcore/requestor.py

```diff
@@ -43,14 +46,17 @@ def __init__(
 
         self.oauth_url = oauth_url
         self.reddit_url = reddit_url
+        self.timeout = timeout
 
     def close(self):
         """Call close on the underlying session."""
         return self._http.close()
 
-    def request(self, *args, timeout=TIMEOUT, **kwargs):
+    def request(self, *args, timeout=None, **kwargs):
         """Issue the HTTP request capturing any errors that may occur."""
         try:
-            return self._http.request(*args, timeout=timeout, **kwargs)
+            return self._http.request(
+                *args, timeout=timeout or self.timeout, **kwargs
```

Comment:
What happens if the passed in timeout is set to `0`?

Reply:
Urllib3 [raises a ValueError](https://github.com/urllib3/urllib3/blob/main/src/urllib3/util/timeout.py#L155-L161) when timeout is nonpositive